paybybank/src/main/java/com/adyen/checkout/paybybank/DefaultPayByBankDelegate.kt

```diff
@@ -0,0 +1,135 @@
+        if (!hasIssuers) {
+            _componentStateFlow.tryEmit(createValidComponentState())
+        } else {
+            _viewFlow.tryEmit(PayByBankComponentViewType)
+        }
+    }
+
+    override fun getPaymentMethodType(): String = paymentMethod.type ?: PaymentMethodTypes.UNKNOWN
+
+    override fun getViewProvider(): ViewProvider = PayByBankViewProvider
+
+    override fun updateInputData(update: PayByBankInputData.() -> Unit) {
+        inputData.update()
+        onInputDataChanged()
+    }
+
+    private fun onInputDataChanged() {
+        val outputData = createOutputData()
+
+        _outputDataFlow.tryEmit(outputData)
+        updateComponentState(outputData)
+    }
+
+    private fun createOutputData() = PayByBankOutputData(
+        selectedIssuer = inputData.selectedIssuer,
+        issuers = filterByQuery()
+    )
+
+    private fun filterByQuery(): List<IssuerModel> = inputData.query?.let { query ->
+        getIssuers().filter { issuerModel ->
+            issuerModel.name.contains(query, ignoreCase = true)
+        }
+    } ?: getIssuers()
+
+    private fun updateComponentState(outputData: PayByBankOutputData) {
+        _componentStateFlow.tryEmit(createComponentState(outputData))
+    }
+
+    private fun createComponentState(outputData: PayByBankOutputData = this.outputData): PaymentComponentState<PayByBankPaymentMethod> {
+        val payByBankPaymentMethod = PayByBankPaymentMethod(
+            type = getPaymentMethodType(), issuer = outputData.selectedIssuer?.id ?: ""
+        )
+
+        val paymentComponentData = PaymentComponentData(
+            paymentMethod = payByBankPaymentMethod
+        )
+
+        return PaymentComponentState(
+            data = paymentComponentData, isInputValid = outputData.isValid, isReady = true
+        )
+    }
+
+    private fun createValidComponentState(): PaymentComponentState<PayByBankPaymentMethod> {
+        val payByBankPaymentMethod = PayByBankPaymentMethod(
+            type = getPaymentMethodType()
+        )
+
+        val paymentComponentData = PaymentComponentData(
+            paymentMethod = payByBankPaymentMethod
+        )
+        return PaymentComponentState(
+            data = paymentComponentData, isInputValid = true, isReady = true
+        )
+    }
+
+    override fun getIssuers(): List<IssuerModel> {
+        return paymentMethod.issuers?.mapToModel() ?: paymentMethod.details.getLegacyIssuers()
+    }
+
+    private fun List<Issuer>.mapToModel(): List<IssuerModel> =
+        this.mapNotNull { issuer ->
+            val (id, name, isDisabled) = issuer
+            if (!isDisabled && id != null && name != null) {
+                IssuerModel(id, name)
+            } else {
+                null
+            }
+        }
+
+    private fun List<InputDetail>?.getLegacyIssuers(): List<IssuerModel> =
+        this.orEmpty()
+            .flatMap { it.items.orEmpty() }
+            .mapNotNull { item ->
+                val (id, name) = item
```

Comment:
I believe you can already do destructuring in the line above:
```Kotlin
.mapNotNull { (id, name) -> 
```

Reply:
Tried it but didn't work.